pkg/storage/storage_put.go

```diff
@@ -24,10 +24,6 @@ type PutInput struct {
 }
 
 func (s *Storage) Put(pi *PutInput) error {
-	// TODO: This is a pretty broad lock. We should find a way to make these locks more selective.
```

Comment:
I haven't seen any finer grained mutex in this PR, is this no longer necessary?

Reply:
I failed to came up with a case when this mutex is needed, neither I found a race condition. It still requires more deep and thorough testing with race detector but to me it seems we can drop the mutex.

Reply:
Decided to revert this: too many invasive changes at a time; should be part of https://github.com/pyroscope-io/pyroscope/pull/685